components/src/jsMain/kotlin/dev/fritz2/components/file.kt

```diff
@@ -0,0 +1,252 @@
+ *
+ * @param styling a lambda expression for declaring the styling as fritz2's styling DSL
+ * @param baseClass optional CSS class that should be applied to the element
+ * @param id the ID of the element
+ * @param prefix the prefix for the generated CSS class resulting in the form ``$prefix-$hash``
+ * @param build a lambda expression for setting up the component itself. Details in [PushButtonComponent]
+ * @return a [Flow] that offers the selected [File]
+ */
+fun RenderContext.file(
+    styling: BasicParams.() -> Unit = {},
+    baseClass: StyleClass? = null,
+    id: String? = null,
+    prefix: String = "file",
+    build: FileSelectionComponent.() -> Unit = {}
+): Flow<File> {
+    var file: Flow<File>? = null
+    val component = FileSelectionComponent().apply(build)
+    (::div.styled(styling, baseClass, id, prefix) {}) {
+        val inputElement = (::input.styled { display { none } }) {
+            type("file")
+            component.accept?.invoke(this)
+            file = changes.events.mapNotNull {
+                domNode.files?.item(0)
+            }.flatMapLatest {
+                domNode.value = "" // otherwise same file can't get loaded twice
+                component.fileReadingStrategy(it)
+            }
+        }.domNode
+        component.renderContext(this, inputElement)
+    }
+    return file!!
+}
+
+/**
+ * This component generates a multiple file selection context.
+ *
+ * In there you can create a button with a label, an icon, the position of the icon and access its events.
+ * For a detailed overview about the possible properties of the button component object itself, have a look at
+ * [PushButtonComponent].
+ *
+ * The [File] function then returns a [Flow] of a [List] of [File]s in order
+ * to combine the [Flow] directly to a fitting _handler_ which accepts a [List] of [File]s:
+ * ```
+ * val textFileStore = object : RootStore<List<String>>(emptyList()) {
+ *   val upload = handle<List<File>>{ _, files -> files.map { it.content } }
+ * }
+ * files {
+ *   accept("text/plain")
+ *   encoding("utf-8")
+ *   button(id = "myFiles") {
+ *     text("Select one or more files")
+ *   }
+ * } handledBy textFileStore.upload
+ * ```
+ * @see PushButtonComponent
+ *
+ * @param styling a lambda expression for declaring the styling as fritz2's styling DSL
+ * @param baseClass optional CSS class that should be applied to the element
+ * @param id the ID of the element
+ * @param prefix the prefix for the generated CSS class resulting in the form ``$prefix-$hash``
+ * @param build a lambda expression for setting up the component itself. Details in [PushButtonComponent]
+ * @return a [Flow] that offers the selected [File]
+ */
+fun RenderContext.files(
+    styling: BasicParams.() -> Unit = {},
+    baseClass: StyleClass? = null,
+    id: String? = null,
+    prefix: String = "file",
+    build: FileSelectionComponent.() -> Unit = {}
+): Flow<List<File>> {
+    var files: Flow<List<File>>? = null
```

Comment:
Same as above: Could this be solved without null? For collections often an empty instance works.